ethpm_types/source.py

```diff
@@ -13,41 +15,72 @@ class Compiler(BaseModel):
 
 
 class Checksum(BaseModel):
-    algorithm: str
-    hash: str
+    """Checksum information about the contents of a source file."""
+
+    algorithm: Algorithm
+    hash: Hex
 
 
 class Source(BaseModel):
+    """Information about a source file included in a Package Manifest."""
+
+    """Array of urls that resolve to the same source file."""
+    urls: List[AnyUrl] = []
+
+    """Hash of the source file."""
     checksum: Optional[Checksum] = None
-    urls: List[str] = []
+
+    """Inlined contract source."""
     content: Optional[str] = None
-    # TODO This was probably done for solidity, needs files cached to disk for compiling
-    # If processing a local project, code already exists, so no issue
-    # If processing remote project, cache them in ape project data folder
+
+    """Filesystem path of source file."""
     installPath: Optional[str] = None
+    # NOTE: This was probably done for solidity, needs files cached to disk for compiling
+    #       If processing a local project, code already exists, so no issue
+    #       If processing remote project, cache them in ape project data folder
+
+    """The type of the source file."""
     type: Optional[str] = None
+
+    """The type of license associated with this source file."""
     license: Optional[str] = None
 
-    def load_content(self):
+    def fetch_content(self) -> str:
         """Loads resource at ``urls`` into ``content``."""
         if len(self.urls) == 0:
-            return
+            raise ValueError("No content to fetch.")
 
         response = urllib.request.urlopen(self.urls[0])
-        self.content = response.read().decode("utf-8")
+        content = response.read().decode("utf-8")
+
+        if self.content and self.content != content:
+            raise ValueError("Content mismatches stored.")
```

Comment:
Is mismatches supposed to be "mismatched" here?

Reply:
```suggestion
            raise ValueError("Content mismatched stored value.")
```

Reply:
19657d8